src/main/haskell/kore/src/Kore/Predicate/Predicate.hs

```diff
@@ -120,14 +117,29 @@ We should consider deleting this and implementing the functionality otherwise.
 wrapPredicate :: StepPattern level variable -> Predicate level variable
 wrapPredicate = GenericPredicate
 
-{- 'unwrapPredicate' wraps a pattern in a GenericPredicate. This should be
-not be used outside of that.
+{- | Unwrap a 'GenericPredicate'.
 
-We should consider deleting this and implementing the functionality otherwise.
--}
+This is intended for predicate evaluation and tests and should not be used
+outside of that.  We should consider deleting this and implementing the
+functionality otherwise.
+
+ -}
 unwrapPredicate :: Predicate level variable -> StepPattern level variable
 unwrapPredicate (GenericPredicate p) = p
 
+{- | Return the 'StepPattern' corresponding to the given 'Predicate'.
+
+All predicates are flexibly-sorted; the sort argument is used to force the
```

Comment:
This is true in in practice, but it's not necessarily true, in theory we could have symbols that only evaluate to top and bottom (@traiansf unless we previously agreed to wrap those in ceil, but I don't remember that; if we did, we should have a design decision doc).

How about prefixing this with "In practice"?

Reply:
I will add the prefix, but note that `Application` patterns can only be turned into a `Predicate` by `wrapPredicate`, which has a note about being unsafe.